python/pyarrow/orc.py

```diff
@@ -175,3 +176,33 @@ def write_table(table, where):
     writer = ORCWriter(where)
     writer.write(table)
     writer.close()
+
+
+def read_table(source, columns=None, filesystem=None):
+    """
+    Read a table from ORC format
+
+    Parameters
+    ----------
+    source : str, pyarrow.NativeFile, or file-like object
+        If a string passed, can be a single file name or directory name. For
+        file-like objects, only read a single file. Use pyarrow.BufferReader to
+        read a file contained in a bytes or buffer-like object.
+    columns : list
+        If not None, only these columns will be read from the file. A column
+        name may be a prefix of a nested field, e.g. 'a' will select 'a.b',
+        'a.c', and 'a.d.e'. If empty, no columns will be read. Note
+        that the table will still have the correct num_rows set despite having
+        no columns.
+    filesystem : FileSystem, default None
+        If nothing passed, paths assumed to be found in the local on-disk
+        filesystem.
```

Comment:
I know this is copied from the parquet read_table docstring, but I am not sure this is fully correct? I suppose passing a URI for eg a remote file also works?

Reply:
It doesn't work like the following.
<img width="1556" alt="Screen Shot 2022-01-18 at 8 38 52 PM" src="https://user-images.githubusercontent.com/9700541/150065159-29158061-f5e7-43b0-9e48-940d982882bd.png">
.

Reply:
We don't yet support http remote filesystem (urls), but do in general support URIs for implemented filesystems such as S3 (eg `s3://bucket/data.orc` _should_ work I think)

Reply:
Got it.